reference/extensions/metadata.rst

```diff
@@ -0,0 +1,279 @@
+    $  $ conan cache path meta/0.1:da39a3ee5e6b4b0d3255bfef95601890afd80709 --folder=metadata
+    <userhome>/.conan2/p/metaf3993f7804ad7/d/metadata
+    # copy my "logs" folder in my cwd to the package metadata folder
+    $ cp -R logs <userhome>/.conan2/p/metaf3993f7804ad7/d/metadata
+    # Now upload the metadata
+    $ conan upload "meta/0.1:*"" -r=remote -c --metadata="*"
+
+
+Downloading metadata
+--------------------
+
+Metadata is not downloaded by default, and it is not downloaded in normal ``install`` operations, when dependencies are downlaoded and installed for usage.
+
+Metadata can be retrieved explicitly by the ``conan download`` command:
+
+.. code-block:: bash
+
+    # download all the metadata from that server
+    $ conan download zlib/1.2.13:* --metadata="*" -r=remote
+    # download the source code (recipe) metadata of all versions of zlib
+    $ conan download zlib/* --metadata="code/*"  -r=remote
+
+
+If we want to download the metadata for a whole dependency graph, it is necessary to use "package-lists":
+
+.. code-block:: bash
+
+    $ conan install . --format=json -r=remote > graph.json
+    $ conan list --graph=graph.json --format=json > pkglist.json
+    # the list will contain the "remote" origin of downloaded packages
+    $ conan download --list=pkglist.json --metadata="*" -r=remote
+
+
+Note that the "package-list" will only contain associated to the "remote" origin the packages that were downloaded. If they were previously in the cache, then, they will not be listed under the "remote" origin and the metadata will not be downloaded.
+
+
+Removing metadata
+-----------------
+
+At the moment it is not possible to remove metadata from the server side using Conan, as the metadata are "additive", it is possible to add new data, but not to remove it (otherwise it would not be possible to add new metadata without downloading first all the previous metadata, and that can be quite inefficient and more error prone, specially sensitive to possible race conditions).
+
+The recommendation to remove metatada from the server side would be to use the tools, web interface or APIs that the server might provide.
+
+
+.. note::
+
+    **Best practices**
+
+    - If you are constantly downloading metadata in many of your ``install`` operations, that might indicate an abuse of the system. Metadata retrieval should be way less common than ``install`` operations.
+    - Package usage shouldn't rely on metadata, if normal consumption of a package makes the metadata mandatory, then this metadata should be part of the package instead.
+    - If there are many small files, it might be necessary to zip those files to avoid excessive upload and download time.
+    - As metadata will not be downloaded by default, it will not be promoted by default in ``conan download`` + ``conan upload`` operations. The recommended way to run a server-server promotion (copy) is using the server tools, and guaranteeing that the metadata folders are copied too.
```

Comment:
Will metadata files also integrate with Artifactory's BuildInfo feature?

Might get tricky if metadata files are added later.

Reply:
Not at this moment. The generation of "modern" BuildInfo has just landed in the Conan extensions https://github.com/conan-io/conan-extensions repository, and will likely require some maturing too. Metadata is a completely new feature, that changes some core management of artifacts by Conan. It is just too early to try to make both things play together. It might not be even possible, as you say, because the capability of being able to add and remove metadata files goes against the declaration stored in a BuildInfo. In other words, the "BuildInfo" is information of built artifacts, but metadata are not build artifacts, but that, metadata, and not covered by BuildInfo. We will see about that later down the road.